modules/schedule_module/jsx/scheduleIndex.js

```diff
@@ -0,0 +1,565 @@
+      editModal: false,
+    });
+  }
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    let result = <td>{cell}</td>;
+    switch (column) {
+    case 'Edit':
+      result = <td><button className="btn btn-default" onClick={() => this.edit(row)}><span className="glyphicon glyphicon-edit"></span> Edit</button></td>;
+      break;
+    case 'Delete':
+      result = <td><button className="btn btn-default" onClick={() => this.deleteid(row['Delete'])}><span className="glyphicon glyphicon-trash"></span> Delete</button></td>;
+      break;
+    case 'Data Entry Status':
+      let css='label label-default';
+      if (cell==='Complete' ) {
+        css='label label-success';
+       }
+      if (cell==='No Data Found' ) {
+        css='label label-danger';
+       }
+      if (cell==='In Progress' || cell==='Not Started') {
+        css='label label-warning';
+       }
+      result = <td><span className={css}>{cell}</span></td>;
+      break;
+    case 'Appointment Type':
+      result = <td>{row['AppointmentTypeName']}</td>;
+      break;
+    }
+
+    return result;
+}
+  renderScheduleFormButton() {
+   if (this.state.editModal) {
+     return (<ButtonElement
+            name="edit"
+            label={
+              <div>
+                <span className="glyphicon glyphicon-plus"/>Edit Appointment
+              </div>
+            }
+            type="submit"
+            buttonClass="btn btn-sm btn-success"
+           />
+     );
+    } else {
+        return (<ButtonElement
+            name="create"
+            label={
+              <div>
+                <span className="glyphicon glyphicon-plus"/>Create Appointment
+              </div>
+            }
+            type="submit"
+            buttonClass="btn btn-sm btn-success"
+           />
+         );
+    }
+  }
+  renderScheduleForm() {
+  let year = new Date();
+  let minYear = year.getFullYear();
+    const title = this.state.editModal ? 'Edit Appointment' : 'Add Appointment';
+    return (
+      <Modal
+        title= {title}
+        onClose={this.closeModal}
+        show={this.state.showModal}
+      >
+        <FormElement
+          Module="schedule"
+          name="addSchedule"
+          id="addScheduleForm"
+          onSubmit={this.handleSubmit}
+          method="POST"
+        >
+          <TextboxElement
+            name="DCCID"
+            label="DCCID"
+            value={this.state.formData.DCCID}
+            required={true}
+            onUserInput={this.setFormData}
+            disabled={this.state.editModal}
+          />
+          <TextboxElement
+            name="PSCID"
+            label="PSCID"
+            value={this.state.formData.PSCID}
```

Comment:
In 'Add Appointment' mode, when I enter the DCCID or the PSCID, the other auto-populates. while this is a cool function, i think it defeats the purpose of having both fields present. I think the reason forms in LORIS ask for both DCCID and PSCID is for double data entry? @driusan what you think?

Reply:
I think it is a good feature, you don't have to load to the access profile to get the PSCID or DCCID.